tools/src/wyvern/target/corewyvernIL/support/ModuleResolver.java

```diff
@@ -59,14 +60,17 @@
     private InterpreterState state;
     private File rootDir;
     private File libDir;
+    private HashMap<String, IExpr> modules;
```

Comment:
What is the purpose of this field?  It seems to duplicate moduleCache.  Is there a way we could just use moduleCache instead?

Reply:
This variable is used to hold the ASTs of the generated modules. We cannot use the module cache for this because all the dependencies of the module being generated might not have been generated. So, we store the ASTs as they get generated in this field. Once all the code has been generated, this field is used along with the specified list of directories for resolving modules.

Reply:
Ok, let's document the purpose of this variable in a comment.  In the medium term, let's also consider whether it makes sense to refactor in a way that more cleanly separates this artifact of code generation from the general module loading.